lib/manager/npm/post-update/index.ts

```diff
@@ -332,6 +332,38 @@ export async function getAdditionalFiles(
   await writeExistingFiles(config, packageFiles);
   await writeUpdatedPackageFiles(config);
 
+  // Store the original npmrc so we can reset it later
+  const npmrcFilePath = upath.join(config.localDir, '.npmrc');
```

Comment:
npm does not check for `.npmrc` recursively so the file needs to be in the same directory as the `package.json`.

Reply:
Instead of doing it based on the package.json I did it based on the package-lock as that seems to be the only time npm install is run.

Reply:
`config.localDir` is not guaranteed to be the location of the `package-lock.json` either. I think this current solution will work if both files are in the root of the repo, but not work for nested.

Reply:
I got this from any time I saw `generateLockFile` it would pass in the cwd path. I'm aware for lerna that you can have nested lock files but I believe in that situation you're still running `npm i` only once from the cwd directory.

I'm curious for my own knowledge, in what case would the local directory that `npm install` is being run not also be the directory the `package-lock` would exist in?

Reply:
You're misunderstanding. `config.localDir` is not always the directory that `package-lock.json` is in. It's always the root directory of the repo, and `package-lock.json` files can be in subdirectories.

Reply:
I think there's a two way miscommunication happening here. I'm not really worried about the lock files themselves more the location that we're running `npm install` or any equivalent of it. 

Based on the [per project config documentation](https://docs.npmjs.com/files/npmrc#per-project-config-file) the npmrc file is expected to be next to the package.json which is also where `npm install` is executed from. So the logic of where I added the code to inject an npmrc file was to track all the locations renovate runs `npm install`.

There is a concern with Lerna where you can have a package.json in any subpackage and technically an npmrc file along with it but honestly I'm not sure why any single Lerna would hook into multiple registries and I'd prefer to handle this as a feature later if someone for some reason has this setup.

As for the actual run of `npm install` (and all equivalents) all seem to be being `exec`-ed in the directory based on the `cwd` parameter being passed into generateLockFile or generateLockFiles. This is the same directory that I'm injecting the npmrc file.

Hopefully that explanation helps and helps give places where you can comment where my thought process is incorrect.